Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeyTable+updateItems.swift

```diff
@@ -203,6 +203,44 @@ public extension AWSDynamoDBCompositePrimaryKeyTable {
         return EventLoopFuture.andAllSucceed(futures, on: self.eventLoop)
     }
     
+    func monomorphicBulkWriteWithFallback<AttributesType, ItemType>(_ entries: [WriteEntry<AttributesType, ItemType>]) -> EventLoopFuture<Void> {
+        // fall back to singel operation if the write entry exceed the statement length limitation
+        var nonBulkWriteEntries: [WriteEntry<AttributesType, ItemType>] = []
+        var bulkWriteEntries: [WriteEntry<AttributesType, ItemType>] = []
+        do {
+            bulkWriteEntries = try entries.compactMap { entry in
+                do {
+                    try self.validateEntry(entry: entry)
+                    return entry
+                } catch SmokeDynamoDBError.statementLengthExceeded {
+                    nonBulkWriteEntries.append(entry)
+                    return nil
+                }
+            }
+        } catch {
+            let promise = self.eventLoop.makePromise(of: Void.self)
+            promise.fail(error)
+            return promise.futureResult
+        }
+            
+        let futures = nonBulkWriteEntries.map { nonBulkWriteEntry -> EventLoopFuture<Void> in
+            switch nonBulkWriteEntry {
+            case .update(new: let new, existing: let existing):
+                return self.updateItem(newItem: new, existingItem: existing)
+            case .insert(new: let new):
+                return self.insertItem(new)
+            case .deleteAtKey(key: let key):
+                return self.deleteItem(forKey: key)
+            case .deleteItem(existing: let existing):
+                return self.deleteItem(existingItem: existing)
+            }
+        }
+        
+        return EventLoopFuture.whenAllSucceed(futures, on: self.eventLoop).flatMap { _ in
```

Comment:
Do we need to wait for all non-bulk writes to complete first? This is not a transaction so I think bulk write can be executed in parallel.

Reply:
Yeah, we can execute them in parallel.